target/linux/ramips/dts/mt7628an_hilink_hlk-7688a-gateway.dts

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "hilink,hlk-7688a-gateway", "hilink,hlk-7688a", "mediatek,mt7628an-soc";
+	model = "Hi-Link HLK-7688A IoT Gateway";
```

Comment:
I'm not a big fan of creating one image per "mode". I'd prefer to create one image hilink,hlk-7688a or hilink,hlk-7688a-iot for both modes.

Reply:
Okay, let me think about this some more.

Reply:
Do you have any tips on the best method to disable the switch when the board is in "IoT Device Mode", i.e. with a single Ethernet (instead of the 1x WAN and 4x LAN of the "IoT Gateway Mode")?

Reply:
While I don't really understand what's that supposed to achieve (why would I have a separate mode which just provides less functionality, I think this is something to be achieved with config. OpenWrt does not aim at providing the exact mirror of the vendor concept _by default_, but just some reasonable configuration. So, you can just flash the official image twice and configure each of them based on the configuration you want to get.

Reply:
By disabling some of the Ethernet ports, the pins can be used for other purposes, such as I2C or SD Card.